server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/ShardRoutingRoleIT.java

```diff
@@ -620,11 +621,20 @@ public void testRefreshOfUnpromotableShards() throws Exception {
 
             indexRandom(true, INDEX_NAME, randomIntBetween(1, 10));
 
-            // Each primary will send a TransportUnpromotableShardRefreshAction to each of the unpromotable replica shards
-            assertThat(
-                refreshUnpromotableActions.get(),
-                is(equalTo((routingTableWatcher.numReplicas - (routingTableWatcher.numIndexingCopies - 1)) * routingTableWatcher.numShards))
-            );
+            int singleRefreshExpectedUnpromotableActions = (routingTableWatcher.numReplicas - (routingTableWatcher.numIndexingCopies - 1))
```

Comment:
Is `numIndexingCopies` always equal to 1?

Reply:
Hi @idegtiarenko , no it's randomly between 1 and 2.

Basically, 1 means that it's the primary, and 2 means there's another indexing replica that is promotable to primary. The remaining replicas are unpromotable/search shards.